packages/vite/src/node/optimizer/index.ts

```diff
@@ -392,7 +392,12 @@ function getDepHash(root: string, config: ResolvedConfig): string {
       plugins: config.plugins.map((p) => p.name),
       optimizeDeps: {
         include: config.optimizeDeps?.include,
-        exclude: config.optimizeDeps?.exclude
+        exclude: config.optimizeDeps?.exclude,
+        esbuildOptions: {
+          plugins: config.optimizeDeps?.esbuildOptions?.plugins?.map(
+            (p) => p.name
+          )
+        }
```

Comment:
I think technically all `esbuildOptions` should be hashed here, with `plugins` having special treatment. But this looks good to me if we want to take one step at a time.